gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -387,6 +393,30 @@ var gpii = fluid.registerNamespace("gpii");
         return logonPromise;
     };
 
+    /**
+     * To re-apply the last environmental login.
+     * This function is used by gpii-app to easily trigger a re-logon of the last environmental login, if for some
+     * reason (e.g. the "reset to standard" function has been activated) the user has become logged out.
+     *
+     * @param {Component} that - An instance of gpii.lifecycleManager.
+     * @param {String} lastEnvironmentalLoginGpiiKey - The GPII key of the last environmental login.
+     * @return {Promise} The promise whose resolved value contains the logon success message and rejected value contains
+     * the fail message.
+     */
+    gpii.lifecycleManager.replayEnvironmentalLogin = function (that, lastEnvironmentalLoginGpiiKey) {
+        var promiseTogo = fluid.promise();
+        var activeGpiiKey = that.getActiveSessionGpiiKey();
+        if (activeGpiiKey && activeGpiiKey !== "noUser") {
+            promiseTogo.reject({
+                isError: true,
+                message: "Replay environmental login ignored because the current active GPII key is not \"noUser\""
```

Comment:
Actually the specification in this case calls for the user's preferences to be reapplied from persistence, assuming that the activeGpiiKey actually is the same one as in the lastEnvironmentalLogon. I think in the case where the key is not "noUser" but different from the lastEnvironmentalLogon we can safely throw an error since this API should not be used in that case.

Reply:
The cheapest way of reapplying the preferences for now might be to log out the user and log them back in again, but we do have enough of a snapshot available in the session in order to achieve this in a less distracting way with a bit more effort.